main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeArrivalAndDepartureTimeValidatorTest.java

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright 2020 Google LLC, MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import org.junit.Before;
+import org.junit.Test;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.StopTimeWithArrivalBeforePreviousDepartureTimeNotice;
+import org.mobilitydata.gtfsvalidator.notice.StopTimeWithDepartureBeforeArrivalTimeNotice;
+import org.mobilitydata.gtfsvalidator.notice.StopTimeWithOnlyArrivalOrDepartureTimeNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableLoader;
+import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+public class StopTimeArrivalAndDepartureTimeValidatorTest {
+    @Mock
+    final GtfsStopTimeTableContainer mockStopTimeTable = mock(GtfsStopTimeTableContainer.class);
+    @InjectMocks
+    final StopTimeArrivalAndDepartureTimeValidator underTest = new StopTimeArrivalAndDepartureTimeValidator();
+
+    @Before
+    public void openMocks() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    public void departureTimeAndArrivalTimeNotProvidedShouldNotGenerateNotice() {
```

Comment:
My interpretation of the spec, given the history from when before the `timepoint` field existed, is that both `arrival_time` and `departure_time` can be blank only if this record is NOT a timepoint. 

Which means that `timepoint` field would need to be empty/missing. Note that the spec says:

>1 or empty - Times are considered exact.

...but to my knowledge this only applies IFF `arrival_time` and `departure_time` are populated - if `arrival_time` and `departure_time` are empty and `timepoint` is empty then the record is NOT a timepoint.

However, I'm not sure the Python validator ever enforced because for a time before `timepoint` existed it became common for producers to provide values for `arrival_time` and `departure_time` for non-timepoint fields even though it was against the spec. @lionel-nj Do you recall the Python validator behavior? This may be the case where if we follow the spec strictly we might be flagging errors on datasets that passed the Python validator.

@timMillet @scmcca Thoughts? The spec could definitely use clarifying in this area for the allowed states of the `arrival_time`, `departure_time`, and `timepoint` fields, including explanation of practice before and after `timepoint` was adopted.

Reply:
IIRC, the python validator only checked if the first and last stop times of a trip had both `arrival_time` and `departure_time` non `null`

Reply:
>IIRC, the python validator only checked if the first and last stop times of a trip had both arrival_time and departure_time non null

Oh, yes, that's something else to consider as well - first and last records always need to have both arrival_time and departure_time.

Reply:
I read the comment several times but I'm not sure to understand accurately the issue. I don't know the practices before the adoption of `timepoint`. Here is what I understand from the spec relatively to `arrival_time` and `departure_time` according to `timepoint`:
- if the `timepoint` field is not provided, all `arrival_time` and `departure_time` values must be provided.
- if the `timepoint` field is provided:
  - if `timepoint=1` on a stop_time record, then the `arrival_time` and `departure_time` values of that record must be provided.
  - if `timepoint=0` or is empty on a stop_time record, then the `arrival_time` and `departure_time` values of that record may be provided, i.e one of them or both can be empty.
  -  first stop_time record and last stop_time record of a trip must have `timepoint=1`, i.e `arrival_time` and `departure_time`.

Don't hesitate to tell me if you want more clarification!

Reply:
@timMillet see https://github.com/google/transit/issues/61 thread for more context on the history of timepoint and some of the current issues.